hazelcast/src/test/java/com/hazelcast/internal/config/override/ExternalClientConfigurationOverrideTest.java

```diff
@@ -42,4 +44,18 @@ public void shouldExtractConfigFromEnv() throws Exception {
         assertEquals("test", config.getInstanceName());
         assertFalse(config.getNetworkConfig().isAutoDetectionEnabled());
     }
+
+    @Test
+    public void shouldHandleConnectionStrategy() throws Exception {
+        ClientConfig config = new ClientConfig();
+        config.getConnectionStrategyConfig()
+          .setReconnectMode(ASYNC)
+          .setAsyncStart(true);
```

Comment:
Shouldn't this be `false` to validate if the override is effective?

Reply:
Good point. However, the most important part here is the validation of the part that's not supposed to be overwritten. But yeah, will change it